docs/bir-spec/src/main/resources/kaitai/bir.ksy

```diff
@@ -177,19 +177,24 @@ types:
         size: value_length
   type_invokable:
     seq:
+      - id: is_any_function
+        type: u1
       - id: param_types_count
         type: s4
+        if: is_any_function == 0
```

Comment:
Typical approach we have used is to define two types of `seq` for each type and switch based on the condition. 
```yaml
  type_invokable:
    seq:
      - id: invokable_kind
        type: u1
        enum: invokable_kind_enum
      - id: cp_info
        type:
          switch-on: tag
          cases:
            'invokable_kind_enum::invokable_kind_any': invokable_rest_parm_info
            'invokable_kind_enum::invokable_kind_single': type_invokable_body
    enums:
      invokable_kind_enum:
        0: invokable_kind_any
        1: invokable_kind_single
  invokable_rest_parm_info:
    seq:
      - id: has_rest_type
        type: u1
      - id: rest_type_cp_index
        type: s4
        if: has_rest_type == 1
  type_invokable_body:
    seq:
      - id: param_types_count
        type: s4
      - id: param_type_cp_index
        type: s4
        repeat: expr
        repeat-expr: param_types_count
      - id: rest_type
        type: invokable_rest_parm_info
      - id: return_type_cp_index
        type: s4
```
This has a small advantage because the generated parser will be more type safe.

Reply:
@manuranga, we probably have to update the BIR version too, right?

Reply:
Yes, we need to change the version as well

Reply:
@pubudu91 s PR https://github.com/ballerina-platform/ballerina-lang/pull/29036#pullrequestreview-608543565 updates this